src/mailmime_write_generic.rs

```diff
@@ -1812,17 +1795,18 @@ pub unsafe fn mailmime_quoted_printable_write_driver(
                                 return r;
                             }
                             start = text.offset(i as isize).offset(1isize);
-                            snprintf(
-                                hexstr.as_mut_ptr(),
-                                6i32 as libc::size_t,
-                                b"=%02X\x00" as *const u8 as *const libc::c_char,
-                                ch as libc::c_int,
+
+                            let raw = format!(
+                                "={}",
+                                hex::encode_upper(&(ch as libc::c_int).to_be_bytes())
```

Comment:
wondering if `={}` pads left with zeros if the value is smaller than 16 (15 should be =0F and not =F)